test/www/jxcore/test/thalicryptomanagertests.js

```diff
@@ -0,0 +1,138 @@
+"use strict";
+
+require('./mockmobile');
+var fs = require('fs');
+var path = require('path');
+var crypto = require('crypto');
+var test = require('tape');
+var cryptomanager = require('../thali/thalicryptomanager');
+
+
+function deleteFolder(fileLocation, file, cb) {
+  if(file) {
+    fs.unlink(file, function(err) {
+      if(err) {
+        console.error('failed to delete the pkcs12 file - err: ', err);
+      }
+      fs.rmdir(fileLocation);
+      cb();
+    });
+  } else {
+    fs.rmdir(fileLocation);
+    cb();
+  }
+}
+
+test('successfully create a new pkcs12 file and return hash value', function (t) {
+  var errorMessage = null,
+      fileLocation = './pkcs12folder';
+  
+  fs.mkdirSync(fileLocation);
+  
+  Mobile.storeGetDocumentsPathReturnArguments(errorMessage, fileLocation);
+  
+  cryptomanager.getPublicKeyHash(function (err, publicKeyHash) {
+    t.equal(err, null);
+
+    var file = path.join(fileLocation, cryptomanager.getPkcs12FileName());
+    fs.readFile(file, function (err, pkcs12Content) {
+      if(err) {
+        console.error('failed to read the pkcs12 file - err: ', err);
+        deleteFolder(fileLocation, file, function() {
```

Comment:
Think of how ridiculously fragile this code is. Wouldn't it be vastly more robust if you had a setup and a tear down function like all test frameworks use? Now, just to confuse things, it turns out Tape doesn't support such an animal. But what then? The first thing a good dev thinks is - it's better to steal than borrow! So you whip our your handy search engine and pretty quickly you find:
https://github.com/isaacs/node-tap - It has a teardown function. Which is good. But it's sorta different than our current test framework. And I don't want to change things that much.
https://www.npmjs.com/package/wrapping-tape - Good license. It's old but it does something simple and I happen to know the dev who wrote it (he's the leader of PouchDB) and he is solid. This requires wrapping the tests in thalicryptomanagertests.js inside the require but that's super easy. 
https://www.npmjs.com/package/redtape - This looks reasonable but it seems fancier than we need, wrapping-tape seems easier.
https://www.npmjs.com/package/tapes - This does the job quite nicely but it would require changing our tests to make them nested. That isn't a big deal, not much work and it has a good license but again it's a bit more involved than I like.

My personal vote would be for wrapping-tape. 

Of course the other option would be to have just written this all yourself. Wrapping-Tape, for example, consists of exactly one file that is 37 lines long. See https://github.com/daleharvey/wrapping-tape/blob/master/wrapping-tape.js so even implementing this ourselves in a slightly slick way would have been no big deal.

But in any case we shouldn't try to do this kind of error handling where we try to find all the exit points and stick in our cleanup code. That is just too error prone. So please give wrapping-tape a try and please clean up the tests.